frappe/integrations/doctype/google_settings/google_settings.py

```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019, Frappe Technologies and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe.model.document import Document
+from frappe import _
+
+class GoogleSettings(Document):
+
+	def validate(self):
+		if self.enable and not (self.client_id and self.client_secret):
```

Comment:
set fields as mandatory

Reply:
Set fields as mandatory only if enable is checked using `frm.set_df_property`, enabling customer to disable google settings without deleting the credentials.